package.json

```diff
@@ -27,7 +27,22 @@
   "scripts": {
     "build": "tsc"
   },
+  "dependencies": {
+    "duplexify": "^4.1.2",
+    "end-of-stream": "^1.4.4",
+    "mqtt-packet": "^7.1.1",
+    "number-allocator": "^1.0.9",
```

Comment:
Do we really need `number-allocator`? I feel like we can implement message ID allocation without taking a dependency. This rewrite is a good opportunity to be really intentional about dependencies.